applications/TrilinosApplication/python_scripts/trilinos_linear_solver_factory.py

```diff
@@ -29,18 +59,131 @@ def _CheckIfTypeIsDeprecated(config):
+    """
+    Constructs and returns a Trilinos linear solver based on the given configuration.
+
+    Args:
+        configuration (KM.Parameters): The configuration parameters for the solver.
+
+    Returns:
+        KratosTrilinos.TrilinosLinearSolverFactory().Create: A Trilinos linear solver.
+
+    Raises:
+        Exception: If the input is not a Kratos Parameters object.
+
+    Example:
+        solver = ConstructSolver(config)
+    """
     if not isinstance(configuration, KM.Parameters):
         raise Exception("input is expected to be provided as a Kratos Parameters object")
 
     _CheckIfTypeIsDeprecated(configuration) # for backwards-compatibility
     _CheckIfSolverIsCompiled(configuration["solver_type"].GetString())
 
     return KratosTrilinos.TrilinosLinearSolverFactory().Create(configuration)
+
+def CreateFastestAvailableDirectLinearSolver():
+    """
+    Creates and returns the fastest available direct linear solver, based on a predefined order and availability.
+
+    TODO: A proper study of speed must be done, particularly depending of system size would be interesting
+
+    In Trilinos, the speed of direct solvers like MUMPS, SuperLU_DIST, KLU, and Basker can depend on various factors including the size and sparsity of the matrix, the architecture of the computer system, and the specific characteristics of the problem being solved. Below are some general observations about these solvers:
+
+    1. MUMPS (MUltifrontal Massively Parallel Sparse Direct Solver):
+    - Parallel solver, handles large problems well.
+    - Can be used on distributed-memory machines.
+    - May have higher memory requirements.
+
+    2. SuperLU_DIST:
+    - Parallel solver, also designed for distributed-memory machines.
+    - Often used for large, sparse linear systems.
+    - Has good performance on a wide range of problems.
+
+    3. KLU:
+    - A serial solver, typically used for smaller problems.
+    - Works well for circuit simulation problems and other problems with a similar structure.
+    - Generally not suitable for large distributed-memory parallel computations.
+
+    4. Basker:
+    - Also a serial solver, suitable for smaller problems.
+    - May not perform as well on larger problems or on distributed-memory systems.
+
+    Here are some considerations for choosing a solver:
+
+    - If you are working on a distributed-memory parallel machine and dealing with large problems, you might want to consider MUMPS or SuperLU_DIST.
+    - For smaller problems or problems with structure similar to circuit simulation problems, KLU might be a good choice.
+    - Basker might be suitable for smaller problems where other solvers are not performing well.
+
+    Returns:
+        ConstructSolver: The fastest available direct linear solver.
+
+    Raises:
+        Exception: If no linear solver could be constructed.
+
+    Example:
+        fast_solver = CreateFastestAvailableDirectLinearSolver()
+    """
+    linear_solvers_by_speed = [
```

Comment:
Are you sure about the order?

I think you are right, but did you test it?

Reply:
I test it in my machine, but accordign to docs depends on several things, like system size and the graph sparsity. I copied/pasted official doc for documentation